### PR TITLE
fix(connlib): Don't roll log files

### LINKED ISSUE #3910: Add opt-in toggle to enable diagnostic log collection
While working through some compliance documentation for the App stores and customer requests, it seems fitting that we should allow users to disable the connlib log uploader.

Logs should still be written to disk as before, just disable the uploader.

Will require a checkbox in GUI clients and env var for CLI clients to set it.

### LINKED ISSUE #4377: NetworkExtension uses NSPrivacyAccessedAPICategoryFileTimestamp causing App Store hurdle
> Hello,
> 
> We noticed one or more issues with a recent submission for TestFlight review for the following app:
> 
> Firezone
> Version 1.0.0
> Build 1711611179
> Although submission for TestFlight review was successful, you may want to correct the following issues in your next submission for TestFlight review. Once you've corrected the issues, upload a new binary to App Store Connect.
> 
> ITMS-91053: Missing API declaration - Your app’s code in the “Firezone” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategoryFileTimestamp. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.
> 
> ITMS-91053: Missing API declaration - Your app’s code in the “PlugIns/FirezoneNetworkExtensioniOS.appex/FirezoneNetworkExtensioniOS” file references one or more APIs that require reasons, including the following API categories: NSPrivacyAccessedAPICategoryFileTimestamp. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, you must include a NSPrivacyAccessedAPITypes array in your app’s privacy manifest to provide approved reasons for these APIs used by your app’s code. For more details about this policy, including a list of required reason APIs and approved reasons for usage, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api.
> 
> Apple Developer Relations

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes #4377 
Closes #3910 

If we decide to implement diagnostic log collection in the future it will be opt-in and use something like Sentry.

### CODE REVIEW

File: rust/connlib/clients/shared/examples/on_demand_rolling.rs
Comment (thomaseizinger) [inline rust/connlib/clients/shared/examples/on_demand_rolling.rs:1]: This entire example can be deleted.

File: rust/linux-client/src/main.rs [hunk 1]
```diff
@@ -97,7 +97,7 @@ fn system_resolvers(dns_control_method: Option<DnsControlMethod>) -> Result<Vec<
 
 #[derive(Clone)]
 struct CallbackHandler {
-    handle: Option<file_logger::Handle>,
+    _handle: Option<file_logger::Handle>,
```
Comment (thomaseizinger) [inline rust/linux-client/src/main.rs]: We should be able to remove this field and store the handle on the stack of `main` to not have it go out of scope.

File: rust/connlib/clients/apple/src/lib.rs [hunk 2]
```diff
@@ -96,7 +96,7 @@ pub struct CallbackHandler {
     // refcount, but there's no way to generate a `Clone` impl that increments the
     // recount. Instead, we just wrap it in an `Arc`.
     inner: Arc<ffi::CallbackHandler>,
-    handle: file_logger::Handle,
+    _handle: file_logger::Handle,
```
Comment (thomaseizinger) [inline rust/connlib/clients/apple/src/lib.rs]: For consistency, we could store this in the `WrappedSession` same as the `Runtime`.

**Comment by jamilbk** [comment]
Tested manually on Apple and Android and it works great. Should behave the same for Linux/Windows.

**Review by thomaseizinger — APPROVED** [review]
Nice! Left some suggestions for more clean up.

File: rust/connlib/clients/apple/src/lib.rs (hunk 2 above)
Comment (jamilbk) [reply rust/connlib/clients/apple/src/lib.rs]: 👍

File: rust/linux-client/src/main.rs (hunk 1 above)
Comment (jamilbk) [reply rust/linux-client/src/main.rs]: 👍

**Pull request merged**